src/ReportAnalysis.ipynb

```diff
@@ -439,13 +439,16 @@
       "source": [
         "# returns all the json file links to the respective inventory reports (by providing the destination container and the rule name along with storage account credentials)\n",
         "def get_json_link_of_reports(storage_account_name, access_key, destination_container, rule_name):\n",
-        "    blob_service = BlockBlobService(storage_account_name, access_key)\n",
-        "    # list the relative path to all the blobs present in the destination container\n",
+        "    # Create a BlobServiceClient object using the account name and access key\n",
+        "    blob_service_client = BlobServiceClient(account_url=f\"https://{storage_account_name}.blob.core.windows.net\", credential=access_key)\n",
         "    try:\n",
```

Comment:
It would be good to let the SDK form the url or generalize it more.

There are partitioned DNS storage accounts whose URLs would look different. It can be taken up in the next iteration of the PR - the current PR can be merged.
https://techcommunity.microsoft.com/blog/azurestorageblog/public-preview-create-additional-5000-azure-storage-accounts-within-your-subscri/3465466